### ISSUE TITLE
Faucet Bricks world

### ISSUE THREAD

**Issue by Xlaits** [MehVahdJukaar/Supplementaries#141]
This happens when you attempt to pour liquid from any container into either nothing or an unsupported block (such as Adorn's sinks). The world is unable to be loaded after this happens, without using AmuletMC to delete the offending faucet. This does not effect any faucet that isn't connected to anything. This also happens when attempting to pour water into a caldron, as far as I am currently aware.

I did test to make sure jar-to-jar interaction works, which it does.

```[23:04:18] [Server thread/ERROR]: Encountered an unexpected exception
net.minecraft.ReportedException: Ticking block entity
        at net.minecraft.server.MinecraftServer.m_5703_(MinecraftServer.java:886) ~[server-1.17.1-20210706.113038-srg.jar%23143!:?]
        at net.minecraft.server.dedicated.DedicatedServer.m_5703_(DedicatedServer.java:334) ~[server-1.17.1-20210706.113038-srg.jar%23143!:?]
        at net.minecraft.server.MinecraftServer.m_5705_(MinecraftServer.java:818) ~[server-1.17.1-20210706.113038-srg.jar%23143!:?]
        at net.minecraft.server.MinecraftServer.m_130011_(MinecraftServer.java:683) ~[server-1.17.1-20210706.113038-srg.jar%23143!:?]
        at net.minecraft.server.MinecraftServer.m_177918_(MinecraftServer.java:258) ~[server-1.17.1-20210706.113038-srg.jar%23143!:?]
        at java.lang.Thread.run(Unknown Source) [?:?]
Caused by: java.lang.NullPointerException: Cannot invoke "net.minecraft.world.level.block.entity.BlockEntity.getCapability(net.minecraftforge.common.capabilities.Capability, net.minecraft.core.Direction)" because "tileBelow" is null
        at net.mehvahdjukaar.supplementaries.block.tiles.FaucetBlockTile.tryFillingBlockBelow(FaucetBlockTile.java:282) ~[supplementaries-1.17.1-0.18.1.jar%23123!:1.17.1-0.18.0]
        at net.mehvahdjukaar.supplementaries.block.tiles.FaucetBlockTile.tryExtract(FaucetBlockTile.java:173) ~[supplementaries-1.17.1-0.18.1.jar%23123!:1.17.1-0.18.0]
        at net.mehvahdjukaar.supplementaries.block.tiles.FaucetBlockTile.tick(FaucetBlockTile.java:75) ~[supplementaries-1.17.1-0.18.1.jar%23123!:1.17.1-0.18.0]
        at net.minecraft.world.level.chunk.LevelChunk$BoundTickingBlockEntity.m_142224_(LevelChunk.java:843) ~[server-1.17.1-20210706.113038-srg.jar%23143!:?]
        at net.minecraft.world.level.chunk.LevelChunk$RebindableTickingBlockEntityWrapper.m_142224_(LevelChunk.java:931) ~[server-1.17.1-20210706.113038-srg.jar%23143!:?]
        at net.minecraft.world.level.Level.m_46463_(Level.java:477) ~[server-1.17.1-20210706.113038-srg.jar%23143!:?]
        at net.minecraft.server.level.ServerLevel.m_8793_(ServerLevel.java:391) ~[server-1.17.1-20210706.113038-srg.jar%23143!:?]
        at net.minecraft.server.MinecraftServer.m_5703_(MinecraftServer.java:882) ~[server-1.17.1-20210706.113038-srg.jar%23143!:?]
        ... 5 more
```

**Comment by MehVahdJukaar** [comment]
I'm aware of this and just haven't had the change to publish a fix since I've still got so much to finish in the current dev branch before I can publish it. it happens every time you activate a faucet that has water but doesn't have a container below and only on 1.17 alpha build. try rolling back one version. Hopefully I'll be able to finish this update soon

**Comment by Xlaits** [comment]
> I'm aware of this and just haven't had the change to publish a fix since I've still got so much to finish in the current dev branch before I can publish it. it happens every time you activate a faucet that has water but doesn't have a container below and only on 1.17 alpha build. try rolling back one version. Hopefully I'll be able to finish this update soon

The mod is part of the Better MC modpack, so I'm not sure if I can roll back. Theoretically it is possible, but if it ever updates, it'll most likely overwrite it.

**Comment by Xlaits** [comment]
> Modpacks aren't locked like that, they are only a collection of mods with an optional compat layer in between (usually configs and custom datapacks). You can edit every modpack very easily, just enable content management in curse forge launcher (under options) and you'll be able to update or add every mod you want. One usually should do that since many modpacks don't update frequently and this leads to users to not have every latest mod version, usually letting many bugs slip in

I'm not 100% sure all launchers are equivalent, but I'm using GDLauncher, which is compatible with CurseForge, FTB, and Technic.

**Comment by MehVahdJukaar** [comment]
so faucet bug was fixed in full release 1.0.0
